src/geometry/aabb.rs

```diff
@@ -1,72 +1,163 @@
-//! An axis-aligned cube.
+//! Axis-aligned box and cube.
 
 use crate::math::sat::{self, ConvexPolyhedron, Intersector, Relation};
 use crate::math::PointCulling;
+use crate::proto;
 use arrayvec::ArrayVec;
-use cgmath::{BaseFloat, Point3, Vector3};
-use collision::{Aabb, Contains};
-use num_traits::Bounded;
+use nalgebra::{Isometry3, Point3, RealField, Unit, Vector3};
+use serde::{Deserialize, Serialize};
 
-pub use collision::Aabb3;
+/// An axis-aligned bounding box.
+#[derive(Debug, PartialEq, Clone, Serialize, Deserialize)]
+pub struct Aabb<S: RealField> {
+    mins: Point3<S>,
+    maxs: Point3<S>,
+}
+
+impl<S: RealField> Aabb<S> {
+    pub fn new(mins: Point3<S>, maxs: Point3<S>) -> Self {
+        Aabb { mins, maxs }
+    }
+
+    pub fn zero() -> Self {
+        Self {
+            mins: Point3::origin(),
+            maxs: Point3::origin(),
+        }
+    }
+
+    pub fn min(&self) -> &Point3<S> {
+        &self.mins
+    }
+
+    pub fn max(&self) -> &Point3<S> {
+        &self.maxs
+    }
+
+    pub fn grow(&mut self, p: Point3<S>) {
+        self.mins = nalgebra::inf(&self.mins, &p);
+        self.maxs = nalgebra::sup(&self.maxs, &p);
+    }
+
+    pub fn contains(&self, p: &Point3<S>) -> bool {
+        nalgebra::partial_le(&self.mins, p) && nalgebra::partial_le(p, &self.maxs)
```

Comment:
`AABB3` of `collision` actually implements it like:
```rust
nalgebra::partial_le(&self.mins, p) && nalgebra::partial_lt(p, &self.maxs)
```
see https://docs.rs/collision/0.20.1/src/collision/volume/aabb/aabb3.rs.html#105-111

Reply:
This would in particular be useful when sharding neighboring bounding boxes.

Reply:
Thanks, changed it.